docs/Style.thy

```diff
@@ -359,6 +359,66 @@ term "
   A" \<comment> \<open>NOT OK: implies this parses as @{text "((A \<and> B) \<longrightarrow> B) \<or> A"}\<close>
 
 
+section \<open>Wrapping/indenting complex rule instantiations, usually as part of @{text "_tac"} methods\<close>
+
+text \<open>
+  For simple instantiations, they can fit it on one line, like so.
+  Notes:
+    * for single-variable instantiations, do not use quotes
+    * no space between @{text "="} and the instantiation or start of quote\<close>
+
+lemma
+  "\<lbrakk>a; b\<rbrakk> \<Longrightarrow> a \<and> b"
+  apply (frule_tac P=a and Q="id b" in conjI) \<comment> \<open>when @{text a} and @{text b} contain bound vars\<close>
```

Comment:
It's not clear to me what the intention of the comment is here. Is it about using the `_tac` form instead of `where`?

Reply:
Yes, that is correct. I want to avoid guiding people towards writing out the rule_tac version when a `fact[where ...]` will suffice. The rule_tac version is less generic, as it applies only to the `_tac`s, but the attribute version has many more uses.

Reply:
I agree with that goal but then I think that the comment needs to be more explicit. As it is the mention of bound variables seems to come out of nowhere and the alternative attribute version isn't mentioned.

However, I'm then struggling to think of a way to keep it short. Would it make sense to add it as a note instead, with something vaguely like
`@{text "_tac"} methods as shown here are generally used when the instantiations contain bound vars, otherwise instantiations with the @{text "[where ...]" attribute are preferred as they are more general.`

Reply:
Ok, tuned, how does the latest commit look to you?

Reply:
One potential typo, otherwise looks good to me!